### ISSUE TITLE
Use TranslateWiki

### ISSUE THREAD

**Issue by FlominatorTM** [FlominatorTM/wikipedia_nearest_photographers#8]
@siebrand  Would this be possible?

**Comment by siebrand** [comment]
Please look into integrating with https://github.com/Krinkle/intuition. That will make it easy on our side, and standardized on your side.

**Comment by FlominatorTM** [comment]
Thanks. Do I get it right, that I have to change the general way my tool retrieves texts with this?

Is there anything from WikiBlame (structure is identical) we could already re-use? Or do I have to start "from scratch"?

**Comment by siebrand** [comment]
I don't think you can use anything from WIkiBlame. Please read the README of Intuition. I couldn't find additional documentation, but maybe @krinkle could point to it, if the README in the repo isn't the best resource.

**Comment by FlominatorTM** [comment]
Sorry for being so reluctant. Regarding documentation https://tools.wmflabs.org/intuition/demo/demo1.php looks promising

Will intuition also work with the tool not running on Labs?

How are you able to provide translations for WikiBlame without intuition support? Can't we use the same mechanism also for this tool?

**Comment by siebrand** [comment]
WikiBlame was configured as a separate product, in a time we didn't have good frameworks. That's something I don't want any more for a product of a few hundred strings or less. Hence my recommendation (or demand) to use Intuition ;).

**Comment by FlominatorTM** [comment]
@siebrand: In the process of integrating Intuition I found the footer mentioned in https://tools.wmflabs.org/intuition/demo/demo5.php which links to labs and apparently won't work with my tool hosted somewhere else.

Now I found the introduction of https://github.com/Krinkle/intuition/wiki/Documentation which states "Intuition is a framework to provide internationalization ( i18n ) for **tools hosted at Tool Labs**."

Therefore I ask you and @Krinkle again: Will intuition also work with the tool not running on Labs?

**Comment by FlominatorTM** [comment]
@siebrand: I think I included it. Can you or @krinkle please check if it is ok like this?

What are the next steps?

**Comment by Krinkle** [comment]
@FlominatorTM Thanks, I've updated the documentation at https://github.com/Krinkle/intuition/wiki/Documentation to reflect that Tool Labs is not a requirement. Intuition is a standalone framework that should work fine in any modern PHP environment.

**Comment by FlominatorTM** [comment]
@Krinkle Thanks for the update and the new parameter. Is there any documentation on how the translation workflow works? How do I make my strings available? From where to I have to retrieve translations?

**Comment by Krinkle** [comment]
@FlominatorTM You're already using the Intuition framework, and have a messages directory that is compatible with its format.

At this point, follow step 3 and 4 from the migration guide https://github.com/Krinkle/intuition/wiki/Migrate#migration.

**Comment by FlominatorTM** [comment]
@Krinkle Step 3 was doable, on step 4 I didn't have a clue what to do with the last four changes of https://github.com/wikimedia/translatewiki/commit/bb2b523fb30  

Also it would be really helpful if the migration guide would be a little more verbose on how to send commits to gerrit.

**Comment by FlominatorTM** [comment]
@Krinkle Thanks for your comments regarding the footer. Can you please give me some hints here as well?

**Comment by FlominatorTM** [comment]
@Krinkle Is there any additional information I can provide you, that might lead to an answer regarding my question from 5th of April?

**Comment by Krinkle** [comment]
@FlominatorTM Are you looking for help with setting up your Gerrit account? It does require more steps than I'd like.

[https://www.mediawiki.org/wiki/Gerrit](https://www.mediawiki.org/wiki/Gerrit/Getting_started)
1. Create a Wikimedia Developer account (LDAP) via [wikitech.wikimedia.org](https://wikitech.wikimedia.org/wiki/Special:UserLogin/signup)
2. Create an SSH key and associate with your account on Gerrit and Wikitech.
3. Clone a repository with the ssh url (or update the remote of an existing repo).
4. Either submit your patch via the git-review cli tool, or use plain Git to [push for review](https://www.mediawiki.org/wiki/Gerrit/Alternatives_to_git-review).

**Comment by FlominatorTM** [comment]
@Krinkle The Gerrit part I got now, thanks. My original problem remains: Your [migration guide](https://github.com/Krinkle/intuition/wiki/Migrate#migration) suggests doing a similiar edit like [this one](https://github.com/wikimedia/translatewiki/commit/bb2b523fb30). I was able to adapt all file changes to my code except the last four (starting with  groups/Intuition/IntuitionAgg.yaml). 

Can you please explain what I'm supposed to change on my code when looking at those last four file diffs?

**Comment by Krinkle** [comment]
@FlominatorTM Can you provide a link to your Gerrit patch so that I can review it and provide feedback?
